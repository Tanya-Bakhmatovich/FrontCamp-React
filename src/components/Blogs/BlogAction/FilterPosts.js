import React from 'react';

export default class FilterPosts extends React.Component {
    constructor() {
        super();
        this.state = {
            value: ''
        };
    }

    handleChange = (e) => {
        this.props.filterBlogs(e.target.value);
    }

    render() {
        return (
            <div className='form-group'>
              <div className='col-auto'>
                  <label> Filter by authors</label>
                      <input
                          id='id-input-filter'
                          type='text'
                          className='form-control'
                          onChange={this.handleChange}
                      />
              </div>
            </div>
            );
        }
    }
