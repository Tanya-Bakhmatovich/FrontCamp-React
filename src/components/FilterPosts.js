import React from 'react';

export default class FilterPosts extends React.Component {

  filter = () => {
    const { posts } = this.props;
    const value = document.getElementById('id-input-filter').value;

    const newPosts = posts.filter(({ author }) => author.indexOf(value) > -1);

    this.props.filterPosts(newPosts);
  }

  render() {

    return (
      <div className={`form-group`}>
          <div className='col-auto'>
              <label> Filter by authors</label>
                  <input
                      id='id-input-filter'
                      type='text'
                      className='form-control'
                      onChange={this.filter}
                  />
          </div>
      </div>
    );
  }
};
