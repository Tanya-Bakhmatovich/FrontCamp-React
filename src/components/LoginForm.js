import React from 'react';
import { Link } from 'react-router-dom';

export default class WelcomePage extends React.Component {
    redirectToBlogs = (e) => {
        e.preventDefault();
        this.props.history.push('/blogs');
    }

    render() {

        return (
            <div>
                <form action='http://localhost:3000/users/login' method='post'>
                    <div className="form-group">
                        <label>Username</label>
                        <input name='username' type='text' className='form-control' />
                    </div>
                    <div className="form-group">
                        <label>Password</label>
                        <input name='password' type='password' className='form-control' />
                    </div>
                    <button onClick={this.redirectToBlogs} id='login-btn' type='submit' className='btn btn-primary'>
                        Log in
                    </button>
                </form>
            </div>
        );
    }
}
