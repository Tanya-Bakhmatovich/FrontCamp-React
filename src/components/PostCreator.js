import React from 'react';

export default class PostCreator extends React.Component {

    createBlock = () => {
        const valueNewPost = {
            title: this.inputTitle.value,
            author: this.inputAuthor.value,
            published: this.inputDate.value,
            text: this.inputText.value,
            id: this.inputId.value,
        }

        this.inputTitle.value = '';
        this.inputAuthor.value = '';
        this.inputDate.value = '';
        this.inputText.value = '';
        this.inputId.value = ''

        this.props.createBlock(valueNewPost);
    }

  render() {
    const { classForm } = this.props;

    return (
      <div className={`form-group ${classForm}`}>
          <div>
              <div className='col-auto'>
                  <label> Title </label>
                      <input ref={input => this.inputTitle = input} type='text' className='form-control'/>
              </div>
              <div className='col-auto'>
                  <label> Author </label>
                      <input ref={input => this.inputAuthor = input} type='text' className='form-control'/>
              </div>
              <div className='col-auto'>
                  <label> Published </label>
                      <input ref={input => this.inputDate = input} type='text' className='form-control'/>
              </div>
              <div className='col-auto'>
                  <label> Text </label>
                      <input ref={input => this.inputText = input} type='text' className='form-control'/>
              </div>
              <div className='col-auto'>
                  <label> Id </label>
                      <input ref={input => this.inputId = input} type='text' className='form-control'/>
              </div>
              <div id='result-creating'></div>
          </div>
          <br/>
          <button
              id='post-blog-button'
              type='button'
              className='btn btn-primary'
              onClick={this.createBlock}
          >
            Create
          </button>
      </div>
    );
  }
}
