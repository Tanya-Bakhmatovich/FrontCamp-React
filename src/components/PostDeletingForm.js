import React from 'react';

export default class PostDeletingForm extends React.Component {

  deletePost = () => {
    const id = document.getElementById('id-input-delete');

    this.props.deletePost(id.value);

    id.value = '';
  }

  render() {

    return (
      <div className={`form-group`}>
          <div className='col-auto'>
              <label> Enter Id for deleting</label>
                  <input id='id-input-delete' type='text' className='form-control'/>
          </div>
          <br/>
          <button
              id='post-blog-button'
              type='button'
              className='btn btn-primary'
              onClick={this.deletePost}
          >
            Delete
          </button>
      </div>
    );
  }
};
