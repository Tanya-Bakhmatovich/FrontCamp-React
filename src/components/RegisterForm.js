import React from 'react';

export default class RegisterForm extends React.Component {

    constructor() {
        super();
        this.state = {
            username: '',
            password: '',
            email: '',
            name: '',
        };
    }

    handleChange = ({ field }) => (e) => {
        const newValue = {};
        newValue[field] = e.target.value;
        this.setState(newValue);
    }

    redirectToLoginForm = (e) => {
//         e.preventDefault();
//         fetch("http://localhost:3000/users", {
//   method: "POST",
//   body: JSON.stringify(this.state),
//   headers: new Headers({
//     'Content-Type': 'application/json'
//   }),
// })
// .then((resp) => {
//   console.log(resp)
// })
// .catch(() => {
//   console.log('failed')
// });
  this.props.history.push('/users/login');
    }

    render() {

        return (
            <div>
                <form id='reg'>
                    <div className="form-group">
                        <label>Username</label>
                        <input onChange={this.handleChange({ field: 'username' })} name='username' type='text' className='form-control' />
                    </div>
                    <div className="form-group">
                        <label>Password</label>
                        <input onChange={this.handleChange({ field: 'password' })} name='password' type='password' className='form-control' />
                    </div>
                    <div className="form-group">
                        <label>Email</label>
                        <input onChange={this.handleChange({ field: 'email' })} name='email' type='text' className='form-control' />
                    </div>
                    <div className="form-group">
                        <label>Name</label>
                        <input onChange={this.handleChange({ field: 'name' })} name='name' type='text' className='form-control' />
                    </div>
                    <button onClick={this.redirectToLoginForm} id='register-btn' type='submit' form='reg' className='btn btn-primary'>
                        Register
                    </button>
                </form>
            </div>
        );
    }
}
