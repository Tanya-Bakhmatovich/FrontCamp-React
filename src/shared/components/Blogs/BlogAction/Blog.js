import React from 'react';

export default class Blog extends React.Component {

    render() {
        const { title, author, published, text, hidden } = this.props.blog;
        const isHiddenClass = hidden ? 'd-none' : '';

        return (
            <div className={`blog-block col-md-4 list-group-item ${isHiddenClass}`}>
                <div className='blog-title'><strong>Title:</strong> {title}</div>
                <div className='blog-author'><strong>Author:</strong> {author}</div>
                <div className='blog-date'><strong>Published:</strong> {published}</div>
                <br/>
                <div className='blog-text'>{text}</div>
            </div>
        );
    }
}
