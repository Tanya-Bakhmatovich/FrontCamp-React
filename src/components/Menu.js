import React from 'react';
import PostCreator from './PostCreator';
import PostDeletingForm from './PostDeletingForm';

export default class Menu extends React.Component {

  constructor() {
      super();
      this.state = {
          disabled: true,
          deleting: false,
      };
  }

  showHideFormAddingPost = () => {
      this.setState(prevState => {
          return {
              disabled: !prevState.disabled,
              deleting: false,
          }
      })
  }

  createBlock = (value) => {
      this.props.onAddingPostHandler(value);
  }

  deletePost = (id) => {
      this.props.onDeletingPostHandler(id);
  }

  changeDeleteState = () => {
      this.setState(prevState => {
          return {
            deleting: !prevState.deleting,
            disabled: true,
          }
      })
  }

  render() {
    const { disabled, deleting } = this.state;
    const { posts } = this.props;
    const classForAddingPostForm = disabled ? "hidden" : "show";
    const classForDeletingPostForm = deleting ? "hidden" : "show";

    return (
        <div>
            <button
                onClick={this.showHideFormAddingPost}
                disabled={!disabled}
                className={`btn btn-primary`}
            >
                Add post
            </button>
            <button
                onClick={this.showHideFormAddingPost}
                disabled={disabled}
                className={`btn btn-primary`}
            >
                Hide form
            </button>
            <button
                onClick={this.changeDeleteState}
                disabled={!posts.length}
                className={`btn btn-primary`}
            >
                Delete post
            </button>
            {!disabled && <PostCreator createBlock={this.createBlock} />}
            {deleting && <PostDeletingForm deletePost={this.deletePost} />}
            <br/>
        </div>
    );
  }
}
