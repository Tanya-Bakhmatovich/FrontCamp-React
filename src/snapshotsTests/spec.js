import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

Enzyme.configure({ adapter: new Adapter() });

import LoginForm from './../shared/LoginForm';
import NavBar from './../shared/NavBar';
import NoMatch from './../shared/NoMatch';
import RegisterForm from './../shared/RegisterForm';
import Blog from '../shared/components/Blogs/BlogAction/Blog';
import BlogUpdatingForm from './../shared/components/Blogs/BlogAction/BlogUpdatingForm';
import FilterPosts from '../shared/components/Blogs/BlogAction/FilterPosts';
import Menu from '../shared/components/Blogs/BlogAction/Menu';
import PostCreator from '../shared/components/Blogs/BlogAction/PostCreator';
import PostDeletingForm from '../shared/components/Blogs/BlogAction/PostDeletingForm';

describe('Component', () => {
  it('NoMatch should match snapshot', () => {
    const wrapper = shallow(<NoMatch />);
    expect(toJson(wrapper)).toMatchSnapshot();
});

it('LoginForm should match snapshot', () => {
  const wrapper = shallow(<LoginForm />);
  expect(toJson(wrapper)).toMatchSnapshot();
});

it('NavBar should match snapshot', () => {
  const wrapper = shallow(<NavBar />);
  expect(toJson(wrapper)).toMatchSnapshot();
});

it('RegisterForm should match snapshot', () => {
  const wrapper = shallow(<RegisterForm />);
  expect(toJson(wrapper)).toMatchSnapshot();
});

it('Blog should match snapshot', () => {
    const blogMock = {
         title: 'title',
         author: 'author',
         published: 'published',
         text: 'text',
         hidden: true
    }
  const wrapper = shallow(<Blog blog={blogMock} />);
  expect(toJson(wrapper)).toMatchSnapshot();
});

it('RegisterForm should match snapshot', () => {
  const wrapper = shallow(<RegisterForm />);
  expect(toJson(wrapper)).toMatchSnapshot();
});

it('BlogUpdatingForm should match snapshot', () => {
  const wrapper = shallow(<BlogUpdatingForm />);
  expect(toJson(wrapper)).toMatchSnapshot();
});

it('FilterPosts should match snapshot', () => {
  const wrapper = shallow(<FilterPosts />);
  expect(toJson(wrapper)).toMatchSnapshot();
});

it('Menu should match snapshot', () => {
    const blogsMock = [{}, {}];
  const wrapper = shallow(<Menu blogs={blogsMock} />);
  expect(toJson(wrapper)).toMatchSnapshot();
});

it('PostCreator should match snapshot', () => {
  const wrapper = shallow(<PostCreator />);
  expect(toJson(wrapper)).toMatchSnapshot();
});

it('PostDeletingForm should match snapshot', () => {
  const wrapper = shallow(<PostDeletingForm />);
  expect(toJson(wrapper)).toMatchSnapshot();
});
});
