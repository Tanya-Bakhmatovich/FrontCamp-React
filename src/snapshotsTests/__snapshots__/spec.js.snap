// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Blog should match snapshot 1`] = `
<div
  className="blog-block col-md-4 list-group-item d-none"
>
  <div
    className="blog-title"
  >
    <strong>
      Title:
    </strong>
     
    title
  </div>
  <div
    className="blog-author"
  >
    <strong>
      Author:
    </strong>
     
    author
  </div>
  <div
    className="blog-date"
  >
    <strong>
      Published:
    </strong>
     
    published
  </div>
  <br />
  <div
    className="blog-text"
  >
    text
  </div>
</div>
`;

exports[`Component BlogUpdatingForm should match snapshot 1`] = `
<div
  className="form-group"
>
  <div
    className="col-auto"
  >
    <label>
       Enter Id for editing
    </label>
    <input
      className="form-control"
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
  <div
    className="col-auto"
  >
    <label>
       Title 
    </label>
    <input
      className="form-control"
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
  <div
    className="col-auto"
  >
    <label>
       Author 
    </label>
    <input
      className="form-control"
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
  <div
    className="col-auto"
  >
    <label>
       Published 
    </label>
    <input
      className="form-control"
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
  <div
    className="col-auto"
  >
    <label>
       Text 
    </label>
    <input
      className="form-control"
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
  <br />
  <button
    className="btn btn-primary"
    id="edit-blog-button"
    onClick={[Function]}
    type="button"
  >
    Update
  </button>
</div>
`;

exports[`Component FilterPosts should match snapshot 1`] = `
<div
  className="form-group"
>
  <div
    className="col-auto"
  >
    <label>
       Filter by authors
    </label>
    <input
      className="form-control"
      id="id-input-filter"
      onChange={[Function]}
      type="text"
    />
  </div>
</div>
`;

exports[`Component LoginForm should match snapshot 1`] = `
<div>
  <Navbar />
  <form
    action="http://localhost:3000/users/login"
    method="post"
  >
    <div
      className="form-group"
    >
      <label>
        Username
      </label>
      <input
        className="form-control"
        name="username"
        onChange={[Function]}
        type="text"
      />
    </div>
    <div
      className="form-group"
    >
      <label>
        Password
      </label>
      <input
        className="form-control"
        name="password"
        onChange={[Function]}
        type="password"
      />
    </div>
    <button
      className="btn btn-primary"
      id="login-btn"
      onClick={[Function]}
      type="submit"
    >
      Log in
    </button>
  </form>
</div>
`;

exports[`Component Menu should match snapshot 1`] = `
<div>
  <button
    className="btn btn-primary"
    disabled={false}
    onClick={[Function]}
  >
    Add post
  </button>
  <button
    className="btn btn-primary"
    disabled={true}
    onClick={[Function]}
  >
    Hide form
  </button>
  <button
    className="btn btn-primary"
    disabled={false}
    onClick={[Function]}
  >
    Delete post
  </button>
  <br />
</div>
`;

exports[`Component NavBar should match snapshot 1`] = `
<ul>
  <li
    key="users/register"
  >
    <NavLink
      activeClassName="active"
      activeStyle={
        Object {
          "fontWeight": "bold",
        }
      }
      ariaCurrent="true"
      to="/users/register"
    >
      Register
    </NavLink>
  </li>
  <li
    key="users/login"
  >
    <NavLink
      activeClassName="active"
      activeStyle={
        Object {
          "fontWeight": "bold",
        }
      }
      ariaCurrent="true"
      to="/users/login"
    >
      Log in
    </NavLink>
  </li>
  <li
    key="blogs"
  >
    <NavLink
      activeClassName="active"
      activeStyle={
        Object {
          "fontWeight": "bold",
        }
      }
      ariaCurrent="true"
      to="/blogs"
    >
      Blogs
    </NavLink>
  </li>
</ul>
`;

exports[`Component NoMatch should match snapshot 1`] = `
<div>
  This page is not exist.
</div>
`;

exports[`Component PostCreator should match snapshot 1`] = `
<div
  className="form-group undefined"
>
  <div>
    <div
      className="col-auto"
    >
      <label>
         Title 
      </label>
      <input
        className="form-control"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div
      className="col-auto"
    >
      <label>
         Author 
      </label>
      <input
        className="form-control"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div
      className="col-auto"
    >
      <label>
         Published 
      </label>
      <input
        className="form-control"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div
      className="col-auto"
    >
      <label>
         Text 
      </label>
      <input
        className="form-control"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div
      className="col-auto"
    >
      <label>
         Id 
      </label>
      <input
        className="form-control"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div
      id="result-creating"
    />
  </div>
  <br />
  <button
    className="btn btn-primary"
    id="post-blog-button"
    onClick={[Function]}
    type="button"
  >
    Create
  </button>
</div>
`;

exports[`Component PostDeletingForm should match snapshot 1`] = `
<div
  className="form-group"
>
  <div
    className="col-auto"
  >
    <label>
       Enter Id for deleting
    </label>
    <input
      className="form-control"
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
  <br />
  <button
    className="btn btn-primary"
    id="post-blog-button"
    onClick={[Function]}
    type="button"
  >
    Delete
  </button>
</div>
`;

exports[`Component RegisterForm should match snapshot 1`] = `
<div>
  <Navbar />
  <form
    id="reg"
  >
    <div
      className="form-group"
    >
      <label>
        Username
      </label>
      <input
        className="form-control"
        name="username"
        onChange={[Function]}
        type="text"
      />
    </div>
    <div
      className="form-group"
    >
      <label>
        Password
      </label>
      <input
        className="form-control"
        name="password"
        onChange={[Function]}
        type="password"
      />
    </div>
    <div
      className="form-group"
    >
      <label>
        Email
      </label>
      <input
        className="form-control"
        name="email"
        onChange={[Function]}
        type="text"
      />
    </div>
    <div
      className="form-group"
    >
      <label>
        Name
      </label>
      <input
        className="form-control"
        name="name"
        onChange={[Function]}
        type="text"
      />
    </div>
    <button
      className="btn btn-primary"
      form="reg"
      id="register-btn"
      onClick={[Function]}
      type="submit"
    >
      Register
    </button>
  </form>
</div>
`;

exports[`Component RegisterForm should match snapshot 2`] = `
<div>
  <Navbar />
  <form
    id="reg"
  >
    <div
      className="form-group"
    >
      <label>
        Username
      </label>
      <input
        className="form-control"
        name="username"
        onChange={[Function]}
        type="text"
      />
    </div>
    <div
      className="form-group"
    >
      <label>
        Password
      </label>
      <input
        className="form-control"
        name="password"
        onChange={[Function]}
        type="password"
      />
    </div>
    <div
      className="form-group"
    >
      <label>
        Email
      </label>
      <input
        className="form-control"
        name="email"
        onChange={[Function]}
        type="text"
      />
    </div>
    <div
      className="form-group"
    >
      <label>
        Name
      </label>
      <input
        className="form-control"
        name="name"
        onChange={[Function]}
        type="text"
      />
    </div>
    <button
      className="btn btn-primary"
      form="reg"
      id="register-btn"
      onClick={[Function]}
      type="submit"
    >
      Register
    </button>
  </form>
</div>
`;
